File: frontend/src/util/Validator.ts
export default class Validator {
	public static instance: Validator = new Validator()
	
	public isCoordinate(coord:number) {
		if(!Number.isInteger(coord)
					|| coord < 0 || coord > 2) {
						throw new Error("Coordinate "+ coord +" is not valid!")
					}
	}
	
	public areCoordinates(...coords: number[]) {
		coords.forEach(coord => this.isCoordinate(coord))
	}

	public isThreeByThree(array: any[][]) {
		if(array.length != 3 || array.some(array2 => array2.length != 3))
			throw new Error("Array must be three by three: "+array)
	}
}

File: backend/logic/src/main/java/dev/pschmalz/ultimate_tic_tac_toe/logic/Validator.java
package dev.pschmalz.ultimate_tic_tac_toe.logic;

public class Validator {
	public static Validator instance = new Validator();
	
	private Validator() {}
	
	public boolean isCoordinateValid(int coord) {
		return 0 <= coord && coord <= 2;
	}
	
	public boolean areCoordinatesValid(int...coords) {
		var valid = true;
		for(var coord : coords) 
			if(!isCoordinateValid(coord))
				valid = false;
		
		return valid;
	}
}
